swapper-org/NodeChain
# Configuring CI/CD for the development environment

# Description



Fixes #11 (issue)
Fixes #19 (issue)

## Dependencies (if any)

- Added [Flake8](https://flake8.pycqa.org/en/latest/) to `requirements.txt`
- Added [PyTest](https://docs.pytest.org/en/6.2.x/) and [PyTest-cov](https://pypi.org/project/pytest-cov/) to `requirements.txt`

## Type of change




# How Has This Been Tested?




### Testing local manually linting and testing
1. Run Lint
2. Run Tests

Process and output [here](https://gist.github.com/swapper-phoenix/f5ff40872ef6ce0fccdaa3a7ffb0e31b).

### Testing CircleCI local CI
1. Log in CircleCI CLI with an account
2. Config the CLI with the personal API key
3. Check if CircleCI configuration file is valid
```shell
$ circleci validate
```
4. Create the `process.yml`
```shell
$ circleci config process .circleci/config.yml > process.yml
```
5. Run the CI
```shell
$ circleci local execute -c process.yml --job build
```

Output [here](https://gist.github.com/swapper-phoenix/6adfb5c9b128df08886a60a42e0ee08e).

### Testing CircleCI production CI
Once we push to any branch, CircleCI run the pipeline automatically.
![imagen](https://user-images.githubusercontent.com/75782378/120215019-87a93280-c235-11eb-8e78-14aeda0a68ca.png)

You can note the same output as running CI locally:
![imagen](https://user-images.githubusercontent.com/75782378/120215126-a90a1e80-c235-11eb-8207-92b558165535.png)


- Test B

**Test Configuration**:
* Operating system (output of `cat /etc/os-release`):
```sh
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.5 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
```
* Kernel version (output of `uname -sr`): Linux 4.15.0-143-generic
* Architecture (output of `uname -m`): x86_64
# Related PR or Docs PR


Docs PR related # 
Other PR related # 

# Good practices to consider
